### ISSUE TITLE
Gwent Homecoming

### ISSUE THREAD

**Issue by jamieadkins95** [GwentCommunityDevelopers/gwent-data#30]
Homecoming will probably make changes to the card data structure. The Open PTR should give us enough time to make the required script changes.

**Comment by jamieadkins95** [comment]
I've added `provision` for all cards, `mulligans` for leaders and `range` for cards with a range limit

**Comment by jamieadkins95** [comment]
`cards.json` will no longer include unreleased cards

**Comment by jamieadkins95** [comment]
Renamed `range` to `reach`

**Comment by teddybee-r** [comment]
Card IDs changed completely and are not correct

**Comment by jamieadkins95** [comment]
I'll investigate tomorrow

It's possible that CDPR have straight up changed the IDs of cards.

**Comment by teddybee-r** [comment]
Card data is incomplete too
it is missing all the PTR cards labeled under 
[MISSING]db_filter_tooltip_expansion_tb
( Deck builder => filter button bottom right )

**Comment by jamieadkins95** [comment]
Cool, thanks for reporting, will look into this tomorrow

**Comment by teddybee-r** [comment]
Leaders are missing:
Eredin, Emhyr, Filavandrel, Crach, Foltest

some keywords are missing for info
{power}, {Maxtargets}

Geralt of Rivia
Deploy: Destroy an enemy with at least {power} power.\n

Geralt: Aard
Deploy.\nMelee: Damage {Maxtargets} enemies by 2 and move them one row up.\n

**Comment by jamieadkins95** [comment]
Fixed tooltips not getting evaluated properly. Note that the PTR only supports English, and the tooltips for other languages are broken.

Fixed missing Thronebreaker cards and missing leaders.

@Ispyy I can't find any cards that have changed ID, do you have any examples?

**Comment by teddybee-r** [comment]
![image](https://user-images.githubusercontent.com/15222466/46582341-5b699380-ca45-11e8-8212-17b17c1bce0e.png)

When I extracted the images from "\Gwent\Gwent_Data\StreamingAssets\bundledassets\cardassets\textures\standard\uber" I get new art IDs

Used: Unity AssetBundleExtractor 2.2beta4

It seems that I failed to mention image IDs specifically

**Comment by jamieadkins95** [comment]
I've just fixed the `ingameArtId` field. Although the `ArtId` doesn't quite line up with the files you are seeing. e.g.:

```
<Template Type="CardTemplate" Id="201660" DebugName="Ruehin" Availability="1" ArtId="1001" AudioId="1213">
    ...
</Template>
```

See `1001` rather than `10010000`

`ingameArtId` just takes the value from `ArtId`, since I don't know that the extra `0000` means and I can't guarantee that it'll produce the right file name. You'll have to append `0000` in your own project.

**Comment by jamieadkins95** [comment]
Categories might not be working properly. Cards only have 1 category at the moment, which isn't right.

**Comment by teddybee-r** [comment]
Card data also isn't up2date.
It might be related to the PTRv2 update they pushed one day later.

I have the correct data in here
`"C:\Users\USERNAME\AppData\LocalLow\CDProjektRED\Gwent\Data\data_definitions"`

This location still has incorrect data (some ability descriptions are wrong) [False Ciri, Lyrian Landsknecht, An Craite Marauder]
`"...\Gwent_PTR\Gwent_Data\StreamingAssets\data_definitions"`

**Comment by jamieadkins95** [comment]
I've fixed the category issue.

See [this commit](https://github.com/GwentCommunityDevelopers/gwent-data/commit/887f320a75f0a27adfff9c614268eb56214dff11).

I haven't got the ptr2 data_definitions file (didn't play the PTR after the first day), and I don't see a need to fix it now that the PTR is over. If you really need that data, feel free to run the script on that newer copy of data_definitions that you have.

Next card data release will be when homecoming releases on the 23rd October.

**Comment by jmxd-public** [comment]
Hello
Could you add the type of card? (unit, leader, spell, artifact)? i believe that data is located in `<Type>` in `Templates.xml`;

Right now you have a "type" but it returns only leader/gold/bronze/silver. Of course that data is also required, but maybe you can rename that to color

Also perhaps the Artist of the card art? it is in `ArtDefinitions.xml` and the "expansion" (you already have this data anyway :D)

Thanks for this script 👍

**Comment by jamieadkins95** [comment]
Good idea. Unfortunately can't rename `type` to `colour` as that will be a breaking change (will make that change in v2), so I'll need to come up with a different name.

I'll also add the Artist name, we had that previously, but CDPR removed it temporarily in a patch months ago.

You can look at the `availability` field for the expansion. `NonOwnable`, `BaseSet`, `Thronebreaker` and `Unmillable`

**Comment by jmxd-public** [comment]
Thanks!

A suggestion for when you will create v2: It would be nice to be able to (more easily) decide the json structure output and the values you want to include.

For example: instead of spread throughout the file, maybe at the bottom list all the collected values that will be in the output. I hope you understand what i mean.

But this is just an idea, not really that important :)

**Comment by jamieadkins95** [comment]
A big focus for v2 for me will be bringing down the size of the output json.

We don't need to include both `info` and `infoRaw` for instance. That should be a command line option.

I'm not sure it'll ever be as configurable as you would like though, and that was never the goal of the project. The goal is to have a standard data format that all developers can rely on, and then you can write your own simple scripts to populate your own projects, a database or whatever you use.

`gwent-data` abstracts away the way Gwent stores card data (which changes a lot), and also stops every dev having to do the same work every patch.

**Comment by jamieadkins95** [comment]
Added `artist` field.

Added `cardType` field. This is terrible since we have `type` already, but I can't think of a better name. May rename it before homecoming. 

For v2 `type` -> `tier` and `cardType` -> `type` to better reflect Gwent terminology

**Comment by jmxd-public** [comment]
Great, thanks!

**Comment by jamieadkins95** [comment]
Initial homecoming release:

https://github.com/GwentCommunityDevelopers/gwent-data-release/tree/homecoming

No card images yet.

**Comment by jamieadkins95** [comment]
Merged homecoming branch #31 . Please open a new issue if there are any problems
